enhancements/hub-in-operator.md

```diff
@@ -0,0 +1,119 @@
+---
+status: proposed
+title: Hub in OpenShift Pipelines
+creation-date: "2021-09-07"
+last-updated: "2021-09-07"
+authors: ["vinamra28"]
+---
+
+# Hub in OpenShift Pipelines Operator
+
+<!-- toc -->
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+- [Design](#design)
+  - [Handling Upgrades](#handling-upgrades)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Tekton Hub provides a central hub for searching and sharing Tekton resources
+across many distributed Tekton catalogs hosted by various organizations and
+teams. Hub currently displays a curated set of community contributed tasks
+from the Community Catalog. It allows resources to be searched by name or
+its “display name”, filtered by categories (cloud, cli, github etc…), and
+rated by users. This document should cover the way how we can ship Tekton Hub
+in a better way so that users can install their own personalized hub within their
+cluster.
+
+## Motivation
+
+With time Hub is gaining more attention by many users and organizations
+and want to deploy the Hub within their own environment. Keeping this
+in mind we should simply the process of deployment so that with less
+effort hub can be installed on anyone's cluster.
+
+### Goals
+
+How to ship Tekton Hub as a part of Tekton/OpenShift Pipelines
+operator.
+
+### Non-Goals
+
+This TEP will not cover how the user is going to expose the service
+to external world via Ingress/Route.
+
+## Requirements
+
+- Images should be made available to the cluster.
+- Target namespace should be already be there in the cluster.
+- Required secrets and configmap should be created in the namespace before creating
+  the resources for Tekton Hub.
+
+## Proposal
+
+Have a CRD in Tekton Operator and a controller/reconciler which can watch
+for resources creation and updation.
```

Comment:
We need a bit more here, like what do we want to be able to configure as part of this CRD ?